### DISCUSSION TITLE
How do I use a client certificate to connect to a git repository?

### DISCUSSION THREAD

**Discussion by hxtk** [fluxcd/flux2#5125] [Q&A] [answered]
There is a GitLab server I would like to use that requires mutually-authenticated TLS using client certificates in addition to access tokens. How do I provide a certificate to Flux for it to use to connect to the Git server? The help output for the bootstrap command describes client certificate options for connecting to the Kubernetes API, but not for connecting to the Git server.

**Comment by stefanprodan** [comment, marked as answer]
There is no support for Git over mTLS, you would need to use `OCIRepository` for mTLS https://fluxcd.io/flux/components/source/ocirepositories/#cert-secret-reference or use SSH with Git.
